File: frontend/src/api/system.ts
import { apiFetch } from "@/api/client"

export type SystemStatus = {
  environment: string
  database: { status: string; schema_status: string; reachable: boolean }
  scheduler: { running: boolean; job_count?: number; jobs?: Record<string, unknown> }
  market_data: { provider: string; cache: Record<string, unknown> }
  scheduled_run_counts: Record<string, number>
  recent_runs: Array<{ run_id: string; job_name: string; status: string; started_at: string; completed_at?: string | null }>
  metrics: Record<string, number>
}

export const systemApi = {
  status: () => apiFetch<SystemStatus>("/system/status"),
}

File: backend/schemas/system.py
"""Operational API response contracts."""

from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel


class RecentRunResponse(BaseModel):
    run_id: UUID
    run_type: str
    portfolio_id: str
    evaluation_date: date
    evaluation_id: UUID | None
    status: str
    provider: str
    started_at: datetime
    completed_at: datetime | None
    duration_ms: float | None
    error_type: str | None


class SystemStatusResponse(BaseModel):
    environment: str
    database: dict[str, str]
    scheduler: dict[str, object]
    market_data: dict[str, object]
    scheduled_run_counts: dict[str, int]
    recent_runs: list[RecentRunResponse]
    metrics: dict[str, object]
